MensaMenuWidget/MensaService.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  MensaService.swift
+//  TUM Campus App
+//
+//  Created by Nikolai Madlener on 12.12.21.
+//  Copyright © 2021 TUM. All rights reserved.
+//
+
+import Foundation
+
+final class MensaService {
+    static let shared = MensaService()
+
+    private init() {}
+    
+    public func getMensaMenu(mensaApiKey: String) async throws -> Menu? {
+        let url = URL(string: "https://tum-dev.github.io/eat-api/" + "\(mensaApiKey)/\(Date().year)/\(Date().weekOfYear).json")!
+        
+        let (data, _) = try await URLSession.shared.data(from: url)
```

Comment:
Using Alamofire might be a bit nicer here.

```suggestion
         let url = EatAPI.menu(location: location, year: Date().year, week: Date().weekOfYear).asURLRequest()
  
          let decoder = JSONDecoder()
          let formatter = DateFormatter.yyyyMMdd
          decoder.dateDecodingStrategy = .formatted(formatter)
  
          AF.request(url).responseDecodable(of: MealPlan.self, decoder: decoder) { response in
              guard let mealPlan = response.value else { return }
              // ...
    }
```